### PR TITLE
POCONC-131: Addition of INR, LCC and RCC to the vitals ribbon

### PR DESCRIPTION
**Pull request by denniskigen**
This commit adds International Normalized Ratio (INR), Left Calf Circumference (LCC) and Right Calf Circumference (RCC) to the vitals ribbon for anti-coagulation patients.

### CODE REVIEW

File: src/app/patient-dashboard/common/todays-vitals/test-todays-vitals.service.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+import { VitalsDatasource } from './vitals.datasource';
+
+@Injectable()
+/**
+ * TestTodaysVitalsService pretends to make real http requests.
+ * implements only as much of TodaysVitalsService as is actually consumed by the app
+*/
+export class TestTodaysVitalsService extends TodaysVitalsService {
+
+  constructor(_vitalsDataSource: VitalsDatasource) {
```
Comment (drizzentic) [inline src/app/patient-dashboard/common/todays-vitals/test-todays-vitals.service.ts]: Avoid the use of underscore prefix for members. Any justification for this?
```suggestion
  constructor(vitalsDataSource: VitalsDatasource) {
```

File: src/app/patient-dashboard/common/todays-vitals/test-todays-vitals.service.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    super(_vitalsDataSource);
```
Comment (drizzentic) [inline src/app/patient-dashboard/common/todays-vitals/test-todays-vitals.service.ts]:
```suggestion
    super(vitalsDataSource);
```

File: src/app/patient-dashboard/common/todays-vitals/test-todays-vitals.service.ts [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  }
+
+  lastResult!: Promise<any>;
+
+  todaysVitals1 = [
```
Comment (drizzentic) [inline src/app/patient-dashboard/common/todays-vitals/test-todays-vitals.service.ts]: why todayVitals1?

File: src/app/patient-dashboard/common/todays-vitals/todays-vitals.component.spec.ts [hunk 4]
```diff
@@ -30,34 +24,248 @@ import {
...
+    label: 'BMI For Age:',
+    name: 'bmiForAge',
+    order: 100,
+    show: true,
+    value: '1'
+  }
+];
 
-  let fixture, el, patientServiceSpy;
+class PatientServiceStub {
```
Comment (drizzentic) [inline src/app/patient-dashboard/common/todays-vitals/todays-vitals.component.spec.ts:35]: can we place the stub in a different file? For instance patient-service-stub.ts

File: src/app/patient-dashboard/common/todays-vitals/todays-vitals.component.spec.ts [hunk 5]
```diff
@@ -30,34 +24,248 @@ import {
...
+      patient: {
+        uuid: 'ce1b2ab4-6cb6-4483-a96d-d4b26ef54f0e'
+      }
+    }
+  ]
+});
 
-  let vitalsService: TodaysVitalsService, patientService: PatientService,
-    fakeAppFeatureAnalytics: AppFeatureAnalytics, component;
+const testVitals = [
```
Comment (drizzentic) [inline src/app/patient-dashboard/common/todays-vitals/todays-vitals.component.spec.ts]: Can we put this testVitals and test patient into a class which will allow reusability of the static data across different tests?

Comment (denniskigen) [reply src/app/patient-dashboard/common/todays-vitals/todays-vitals.component.spec.ts]: Sure. Extracting all the mock data into a mock file has exposed a frailty I had in my tests where I hadn't considered that hard-coding a value for the encounter datetime would mean there would be no vitals for the next day and/or subsequent days.

File: src/app/patient-dashboard/common/todays-vitals/test-todays-vitals.service.ts (hunk 3 above)
Comment (denniskigen) [reply src/app/patient-dashboard/common/todays-vitals/test-todays-vitals.service.ts]: Ideally I'd wanted to have two mock vitals - moving all the mock data into its own file solves this

File: src/app/patient-dashboard/common/todays-vitals/test-todays-vitals.service.ts (hunk 1 above)
Comment (denniskigen) [reply src/app/patient-dashboard/common/todays-vitals/test-todays-vitals.service.ts]: The constructor required that I have a different vitalsDataSource singleton than the one in the parent class I'd extended. Prefixing this private instance of vitalsDataSource with an underscore differentiated it with the one in the parent class and therefore resolved the error. More recently however, getting rid of TestTodaysVitalsService and replacing it with a simple stub resolves this.

File: src/app/patient-dashboard/common/todays-vitals/todays-vitals.component.spec.ts (hunk 4 above)
Comment (denniskigen) [reply src/app/patient-dashboard/common/todays-vitals/todays-vitals.component.spec.ts:35]: This being such a small object I don't think it merits moving into its own file

Comment (drizzentic) [reply src/app/patient-dashboard/common/todays-vitals/todays-vitals.component.spec.ts:35]: Wouldn't we have other services in the future which might require us to stub?

Comment (denniskigen) [reply src/app/patient-dashboard/common/todays-vitals/todays-vitals.component.spec.ts:35]: Possibly. But in this scenario I don't feel that these three stubs are anywhere near large enough to merit breaking them out into their own individual files.

**Review by drizzentic — APPROVED** [review]
Looks good to me

**Review by enyachoke — APPROVED** [review]

**Pull request merged**